### PR TITLE
[ROCm] Ship rocprofiler-sdk 1.3.2 in Dockerfile.rocm_base to fix torch.profiler traces

### PR DESCRIPTION
**Pull request by Rohan138**
# [ROCm] Ship rocprofiler-sdk 1.3.2 in Dockerfile.rocm_base to fix torch.profiler HIP-graph traces

## Summary
The ROCm ≤ 7.2.3 base image (`rocm/dev-ubuntu-22.04:7.2.3-complete`) ships **rocprofiler-sdk 1.1.0**.
Since torch 2.12, PyTorch's Kineto GPU backend links `librocprofiler-sdk` (previously roctracer).
Under `torch.profiler`, rocprofiler-sdk **1.1.0** blows up `hipGraphLaunch` and disables the HIP
graph fast path, so profiled decode collapses GPU occupancy and profiled latency balloons —
making vLLM profiles unusable/misleading on gfx942/gfx950.

This is a **rocprofiler-sdk bug fixed in 1.3.2**. This PR builds rocprofiler-sdk 1.3.2
(`ROCm/rocm-systems @ 2b22ab0`) in a dedicated stage and overlays it onto `/opt/rocm` in the
final base image. torch/Kineto links `librocprofiler-sdk.so.1` by soname, so swapping the
runtime lib is sufficient — **no torch rebuild**, `build_pytorch` is untouched.

Self-removing: drop the stage + COPY once `BASE_IMAGE` ships rocprofiler-sdk ≥ 1.3.2 (ROCm 7.14,
tracked in #48826).

## Not a duplicate
Checked open PRs/issues (`rocprofiler`, `rocprofiler-sdk`, `kineto`, `roctracer`): no PR changes
the base image's rocprofiler-sdk. Related: #48826 (ROCm 7.14 migration — the eventual superseder)
and closed/stale #39162 (same `rocprofiler_configure in libtorch_cpu.so` observation, no fix).

## Evidence (gfx942 / MI300X, torch 2.12, llama-3.1-8B, `bench latency --profile`)
Measured on an image produced by this exact change (`docker build` of the new stage +
`COPY`/`ldconfig`, then the vLLM layer):
| rocprofiler-sdk | GPU occupancy | gaps >1ms | hipGraphLaunch avg |
|---|---|---|---|
| 1.1.0 (base) | 39.9% | 578 | 16.6 ms |
| **1.3.2 (this PR)** | **97.8%** | **2** | **893 µs** |

Microbench (HIP graph replay under profiler): `hipGraphLaunch` 380 µs → 56 µs (≈ roctracer's
50 µs). ROCm 7.14 (native rocprofiler-sdk 1.3.2) independently shows the healthy result (98.6%
occupancy at e2e), confirming the rocprofiler-sdk version is the fix.

## Test commands
```
# Base image build (surgical stage builds in-context; rest of the base is unchanged):
docker build -f docker/Dockerfile.rocm_base -t <base>:rocprofsdk-1.3.2 .
docker run --rm <base>:rocprofsdk-1.3.2 bash -c \
  'ldd $(python -c "import torch,pathlib;print(pathlib.Path(torch.__file__).parent/\"lib\")")/libtorch_cpu.so | grep rocprofiler-sdk; \
   cat /app/versions.txt | grep ROCPROFILER_SDK'
# expect: librocprofiler-sdk.so.1 -> ...1.3.2

# llama profile (on a vLLM image built from this base):
VLLM_TORCH_PROFILER_DIR=/tmp/prof vllm bench latency --model meta-llama/Llama-3.1-8B-Instruct \
  -tp 1 --batch-size 32 --input-len 512 --output-len 128 --num-iters 3 \
  --profile --profiler-config.profiler=torch --profiler-config.torch_profiler_dir=/tmp/prof
# expect: profiled GPU occupancy ~99%, hipGraphLaunch sub-ms (vs ~16ms on 1.1.0)
```

## Notes
- Adds a ~few-minute rocprofiler-sdk compile to base-image builds (uses `sccache` when
  `USE_SCCACHE=1`, matching the existing stages).
- Pinned commit is overridable via `--build-arg ROCPROFILER_SDK_COMMIT=...`.

## AI assistance
Developed with AI assistance (Claude): the investigation, the Dockerfile stage, and the
build + llama-profile validation above were AI-run on the submitter's MI300X box. Opened as
a **draft** pending the submitter's final line-by-line review before marking ready.

### CODE REVIEW

**Comment by mawong-amd** [comment]
Concern: we might also want to sync https://github.com/vllm-project/vllm/blob/7f6432cc0daf97c5b9a2df141641c595a766ff40/.buildkite/scripts/rocm/refresh-base-image.sh so it doesn't keep rebuilding the `rocm_base` image (and therefore also the `ci_base` image). cc @AndreasKaratzas

**Comment by AndreasKaratzas** [comment]
@Rohan138 I checked the ROCm bake/cache flow locally. The cache identity already includes the complete `Dockerfile.rocm_base`, the resolved upstream base-image digest, and the sccache mode. Therefore, the new rocprofiler commit changes the content hash once, and subsequent runs reuse it. I did notice one NIT in the Dockerfile: the trailing

```sh
&& find ... -exec strip ... + 2>/dev/null || true
```

can probably mask failures from the preceding CMake configure/build/install commands, not only failures from `strip`. Would it be better to remove the broad `|| true`, use `cmake --install build --prefix /staging --strip`, or scope the ignored failure specifically around the `find` command?

**Comment by Rohan138** [comment]
Good catch — `... || true` bound to the whole `&&` chain, so a cmake configure/build/install failure would have been swallowed too. Switched to `cmake --install build --prefix /staging --strip` (dropping the separate `find … strip … || true`), so any build/install failure now propagates and the image still ships stripped libs. Rebuilt the stage to confirm it builds and the stripped `librocprofiler-sdk.so.1.3.2` still loads under torch.

Thanks also for confirming the base-image cache picks up the content change once and reuses it afterward — that matches what I saw, so no `refresh-base-image.sh` change needed.

**Comment by Fangzhou-Ai** [comment]
@AndreasKaratzas  can we merge this PR?

**Review by AndreasKaratzas — APPROVED** [review]
